pkgs/by-name/du/duckstation/package.nix

```diff
@@ -1,35 +1,43 @@
-{ lib
-, stdenv
-, fetchFromGitHub
-, SDL2
-, cmake
-, cubeb
-, curl
-, extra-cmake-modules
-, libXrandr
-, libbacktrace
-, makeWrapper
-, ninja
-, pkg-config
-, qtbase
-, qtsvg
-, qttools
-, qtwayland
-, substituteAll
-, vulkan-loader
-, wayland
-, wrapQtAppsHook
+{
+  lib,
+  SDL2,
+  callPackage,
+  cmake,
+  cubeb,
+  curl,
+  extra-cmake-modules,
+  fetchFromGitHub,
+  libXrandr,
+  libbacktrace,
+  libwebp,
+  makeWrapper,
+  ninja,
+  pkg-config,
+  qt6,
+  stdenv,
+  substituteAll,
+  vulkan-loader,
+  wayland,
 }:
-
+let
+  shaderc-patched = callPackage ./shaderc-patched.nix { };
```

Comment:
Now no one can override `shaderc` without using `overrideAttrs` and modifying the `buildInputs` - something that would have been possible if you had used `finalAttrs.passthru.shaderc` as I suggested back then. However, that's not mandatory, and many packages work like this and probably barely anyone complains, so it's not too bad.

Reply:
As I said before, shaderc works like a vendored lib - but instead of pulling the whole code, they just distribute a patch.